File: csharp/src/main/CsharpDotNet2/spoonacular/Model/InlineResponse2001.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace spoonacular.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class InlineResponse2001 {
    /// <summary>
    /// Gets or Sets Id
    /// </summary>
    [DataMember(Name="id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "id")]
    public int? Id { get; set; }

    /// <summary>
    /// Gets or Sets Image
    /// </summary>
    [DataMember(Name="image", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "image")]
    public string Image { get; set; }

    /// <summary>
    /// Gets or Sets ImageType
    /// </summary>
    [DataMember(Name="imageType", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "imageType")]
    public string ImageType { get; set; }

    /// <summary>
    /// Gets or Sets Likes
    /// </summary>
    [DataMember(Name="likes", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "likes")]
    public int? Likes { get; set; }

    /// <summary>
    /// Gets or Sets MissedIngredientCount
    /// </summary>
    [DataMember(Name="missedIngredientCount", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "missedIngredientCount")]
    public int? MissedIngredientCount { get; set; }

    /// <summary>
    /// Gets or Sets MissedIngredients
    /// </summary>
    [DataMember(Name="missedIngredients", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "missedIngredients")]
    public List<RecipesFindByIngredientsMissedIngredients> MissedIngredients { get; set; }

    /// <summary>
    /// Gets or Sets Title
    /// </summary>
    [DataMember(Name="title", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "title")]
    public string Title { get; set; }

    /// <summary>
    /// Gets or Sets UnusedIngredients
    /// </summary>
    [DataMember(Name="unusedIngredients", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "unusedIngredients")]
    public List<Object> UnusedIngredients { get; set; }

    /// <summary>
    /// Gets or Sets UsedIngredientCount
    /// </summary>
    [DataMember(Name="usedIngredientCount", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "usedIngredientCount")]
    public decimal? UsedIngredientCount { get; set; }

    /// <summary>
    /// Gets or Sets UsedIngredients
    /// </summary>
    [DataMember(Name="usedIngredients", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "usedIngredients")]
    public List<RecipesFindByIngredientsMissedIngredients> UsedIngredients { get; set; }

    /// <summary>
    /// Gets or Sets 
    /// </summary>
    [DataMember(Name="", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "")]
    public string  { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class InlineResponse2001 {\n");
      sb.Append("  Id: ").Append(Id).Append("\n");
      sb.Append("  Image: ").Append(Image).Append("\n");
      sb.Append("  ImageType: ").Append(ImageType).Append("\n");
      sb.Append("  Likes: ").Append(Likes).Append("\n");
      sb.Append("  MissedIngredientCount: ").Append(MissedIngredientCount).Append("\n");
      sb.Append("  MissedIngredients: ").Append(MissedIngredients).Append("\n");
      sb.Append("  Title: ").Append(Title).Append("\n");
      sb.Append("  UnusedIngredients: ").Append(UnusedIngredients).Append("\n");
      sb.Append("  UsedIngredientCount: ").Append(UsedIngredientCount).Append("\n");
      sb.Append("  UsedIngredients: ").Append(UsedIngredients).Append("\n");
      sb.Append("  : ").Append().Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: dart/com/spoonacular/client/model/inline_response2001.dart
part of openapi.api;

class InlineResponse2001 {
  
  int id = null;
  
  String image = null;
  
  String imageType = null;
  
  int likes = null;
  
  int missedIngredientCount = null;
  
  List<RecipesFindByIngredientsMissedIngredients> missedIngredients = [];
  
  String title = null;
  
  List<Object> unusedIngredients = [];
  
  num usedIngredientCount = null;
  
  List<RecipesFindByIngredientsMissedIngredients> usedIngredients = [];
  
  String  = null;
  InlineResponse2001();

  @override
  String toString() {
    return 'InlineResponse2001[id=$id, image=$image, imageType=$imageType, likes=$likes, missedIngredientCount=$missedIngredientCount, missedIngredients=$missedIngredients, title=$title, unusedIngredients=$unusedIngredients, usedIngredientCount=$usedIngredientCount, usedIngredients=$usedIngredients, =$, ]';
  }

  InlineResponse2001.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['id'] == null) {
      id = null;
    } else {
          id = json['id'];
    }
    if (json['image'] == null) {
      image = null;
    } else {
          image = json['image'];
    }
    if (json['imageType'] == null) {
      imageType = null;
    } else {
          imageType = json['imageType'];
    }
    if (json['likes'] == null) {
      likes = null;
    } else {
          likes = json['likes'];
    }
    if (json['missedIngredientCount'] == null) {
      missedIngredientCount = null;
    } else {
          missedIngredientCount = json['missedIngredientCount'];
    }
    if (json['missedIngredients'] == null) {
      missedIngredients = null;
    } else {
      missedIngredients = RecipesFindByIngredientsMissedIngredients.listFromJson(json['missedIngredients']);
    }
    if (json['title'] == null) {
      title = null;
    } else {
          title = json['title'];
    }
    if (json['unusedIngredients'] == null) {
      unusedIngredients = null;
    } else {
      unusedIngredients = Object.listFromJson(json['unusedIngredients']);
    }
    if (json['usedIngredientCount'] == null) {
      usedIngredientCount = null;
    } else {
          usedIngredientCount = json['usedIngredientCount'];
    }
    if (json['usedIngredients'] == null) {
      usedIngredients = null;
    } else {
      usedIngredients = RecipesFindByIngredientsMissedIngredients.listFromJson(json['usedIngredients']);
    }
    if (json[''] == null) {
       = null;
    } else {
           = json[''];
    }
  }

  Map<String, dynamic> toJson() {
    Map <String, dynamic> json = {};
    if (id != null)
      json['id'] = id;
    if (image != null)
      json['image'] = image;
    if (imageType != null)
      json['imageType'] = imageType;
    if (likes != null)
      json['likes'] = likes;
    if (missedIngredientCount != null)
      json['missedIngredientCount'] = missedIngredientCount;
    if (missedIngredients != null)
      json['missedIngredients'] = missedIngredients;
    if (title != null)
      json['title'] = title;
    if (unusedIngredients != null)
      json['unusedIngredients'] = unusedIngredients;
    if (usedIngredientCount != null)
      json['usedIngredientCount'] = usedIngredientCount;
    if (usedIngredients != null)
      json['usedIngredients'] = usedIngredients;
    if ( != null)
      json[''] = ;
    return json;
  }

  static List<InlineResponse2001> listFromJson(List<dynamic> json) {
    return json == null ? new List<InlineResponse2001>() : json.map((value) => new InlineResponse2001.fromJson(value)).toList();
  }

  static Map<String, InlineResponse2001> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, InlineResponse2001>();
    if (json != null && json.isNotEmpty) {
      json.forEach((String key, dynamic value) => map[key] = new InlineResponse2001.fromJson(value));
    }
    return map;
  }
}

